ext/TrixiConvexECOSExt.jl

```diff
@@ -67,6 +67,48 @@ function stability_polynomials!(pnoms, consistency_order, num_stage_evals,
     end
 end
 
+# Specialized form of the stability polynomials for fourth-order PERK schemes.
+function stability_polynomials_PERK4!(pnoms, num_stage_evals,
+                                      normalized_powered_eigvals,
+                                      gamma, dt, cS3)
+    num_eig_vals = length(pnoms)
+
+    # Constants arising from the particular form of Butcher tableau chosen for the 4th order PERK methods
+    k1 = 0.001055026310046423 / cS3
+    k2 = 0.03726406530405851 / cS3
+    # Note: `cS3` = c_{S-3} is in principle free, while the other abscissae are fixed to 1.0
+
+    # Initialize with zero'th order (z^0) coefficient
+    for i in 1:num_eig_vals
+        pnoms[i] = 1.0
+    end
+
+    # First `consistency_order` = 4 terms of the exponential
+    for k in 1:4
+        for i in 1:num_eig_vals
+            pnoms[i] += dt^k * normalized_powered_eigvals[i, k]
+        end
+    end
+
```

Comment:
Same idea as my comment below, these lines are serving the same functionality as lines 44-45
A function that I would propose to avoid writing the same code twice would be

```
@inline function coeff_free_terms_stability_polynomials!(polynom, num_eig_vals, consistency_order)
    for i in 1:num_eig_vals
        pnoms[i] = 1.0
    end
    for k in 1:consistency_order
        for i in 1:num_eig_vals
            pnoms[i] += normalized_powered_eigvals_scaled[i, k]
        end
    end
end
```

The function's name probably needs rework since for order 4, I see that the term z^5 is also not dependent on the free coefficients.